### PR TITLE
liquid-crowdloan

### LINKED ISSUE #2462: Liquid Crowdloan redeem pallet
TBD

### PR DESCRIPTION
**Pull request by xlc**
closes #2462 

TODOs:
- [x] XCM to transfer DOT into the pallet account
- [x] tests
- [x] benchmarking & weights

### CODE REVIEW

File: runtime/acala/src/lib.rs [hunk 1]
```diff
@@ -1650,6 +1655,17 @@ impl nutsfinance_stable_asset::Config for Runtime {
 	type EnsurePoolAssetId = EnsurePoolAssetId;
 }
 
+impl module_liquid_crowdloan::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Currencies;
+	type LiquidCrowdloanCurrencyId = GetLiquidCurrencyId;
+	type RelayChainCurrencyId = GetStakingCurrencyId;
+	type PalletId = LiquidCrowdloanPalletId;
+	type GovernanceOrigin = EnsureRoot<AccountId>;
```
Comment (xlc) [inline runtime/acala/src/lib.rs:1664]: ensure root or more than half council

File: modules/relaychain/src/lib.rs [hunk 2]
```diff
@@ -66,20 +68,48 @@ pub enum StakingCall {
...
+	),
+}
+
+// Same to `Polkadot` and `Kusama` runtime `Lookup` config.
+pub type RelayChainLookup = AccountIdLookup<AccountId, ()>;
+
+/// `pallet-proxy` calls.
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum ProxyCall<RelayChainCall> {
+	/// `proxy(real, force_proxy_type, call)` call.
```
Comment (xlc) [inline modules/relaychain/src/lib.rs:90]: add a comment explain `Option<()>` i.e. we don't support ProxyType

File: runtime/acala/src/lib.rs [hunk 3]
```diff
@@ -1650,6 +1653,22 @@ impl nutsfinance_stable_asset::Config for Runtime {
...
 
+parameter_types!(
+	// Crowdloan vault address: `132zsjMwGjNaUXF5XjUCDs2cDEq9Qao51TsL9RSUTGZbinVK`
+	pub CrowdloanVault: AccountId = AccountId::from(hex_literal::hex!("59fe89295c2e57d7b4d4d8be9e00a3802e513703ab4b5b424ed0a646e899d3c9"));
+);
+
+impl module_liquid_crowdloan::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Currencies;
+	type LiquidCrowdloanCurrencyId = GetLiquidCurrencyId;
```
Comment (zjb0807) [inline runtime/acala/src/lib.rs:1664]:
```suggestion
	type LiquidCrowdloanCurrencyId = LCDOT;
```

File: modules/liquid-crowdloan/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+			T::Currency::transfer(T::RelayChainCurrencyId::get(), &Self::account_id(), &who, amount)?;
+
+			Self::deposit_event(Event::Redeemed { amount });
+
+			Ok(())
+		}
+
+		/// Send an XCM message to cross-chain transfer DOT from relay chain crowdloan vault to
+		///  liquid crowdloan module account.
```
Comment (zjb0807) [inline modules/liquid-crowdloan/src/lib.rs:107]:
```suggestion
		/// liquid crowdloan module account.
```

File: modules/liquid-crowdloan/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+pub mod module {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = CurrencyId, Balance = Balance>;
+
+		/// Liquid crowdloan currency Id, i.e. LDOT for Polkadot.
```
Comment (xlc) [inline modules/liquid-crowdloan/src/lib.rs:51]:
```suggestion
		/// Liquid crowdloan currency Id, i.e. LCDOT for Polkadot.
```

**Pull request merged**